### PR TITLE
add note on binbreaking

### PR DESCRIPTION
**Pull request by jmcardon**

### CODE REVIEW

**Comment by jmcardon** [comment]
Good point. I can add a note on that. Major versions = no bincompat. Minor versions = bincompat.

`tsec-0.1.0` shouldn't be bincompat with `tsec-0.2.0`, but `tsec-0.1.1` should be with `0.1.2`

**Comment by ChristopherDavenport** [comment]
Isn't `0.0.1` patch, `0.1.0` minor, and `1.0.0` major? I feel I read that `0.1.0` was a major version which was odd to me.

File: README.md [hunk 1]
```diff
@@ -20,6 +20,13 @@ For the current progress, please refer to the [RoadMap](https://github.com/jmcar
 For version changes and additions, including breaking changes, see either [release notes](https://github.com/jmcardon/tsec/releases)
 or the [Version Changes](https://github.com/jmcardon/tsec/wiki/Version-Changes) page.
 
+### Note on milestones:
+All `x.x.x-Mx` releases are milestone releases. Thus, we do not guarantee binary compatibility or no api-breaking until
+a concrete version(i.e `0.0.1`). We aim to keep userland-apis relatively stable, but 
+internals shift as we find better/more performant abstractions.
+
+We will guarantee compat between minor versions (i.e 0.0.1 => 0.0.2) but not major versions (0.0.1 => 0.1.0)
```
Comment (ChristopherDavenport) [inline README.md]: `s/minor/patch/g` `s/major/minor/g`

**Comment by hrhino** [comment]
That's semver, yes. I'm not a huge fan of it as it is. I prefer to keep the leftmost digit for redesigns / huge changes or whatever: things that would require more than minor source changes. Having to bump that number just to break bincompat in any way seems unpleasant to me.

File: README.md (hunk 1 above)
Comment (hrhino) [reply README.md]: good catch. reviewing on the train considered not that effective.

**Comment by ChristopherDavenport** [comment]
All good, just thought if we were not going to keep semver we should note that explicitly, not merely describing how we are going to do it differently.

**Comment by hrhino** [comment]
That's reasonable, although I think the strict definition of "semantic versioning" isn't too common in java/scala circles.

**Comment by jmcardon** [comment]
@ChristopherDavenport you have a good point actually...

We should note on this. I have the leftmost digit in my mind as the "deploy the nukes, we have a huge change now" button, rather then semver.

**Comment by hrhino** [comment]
That's because we're scala devs, and scala `3.x.y` involves a complete rewrite.

Also scalaz devs, for whom scalaz `8.x.y` involves a complete rewrite :P

File: README.md [hunk 2]
```diff
@@ -20,6 +20,21 @@ For the current progress, please refer to the [RoadMap](https://github.com/jmcar
...
+Our Notation for versions is:
+```
+X.X.X
+^ ^ ^____Minor
+| |______Major
+|________Complete redesign (i.e scalaz 7 vs 8)  
+```
+
+All `x.x.x-Mx` releases are milestone releases. Thus, we do not guarantee binary compatibility or no api-breaking until
+a concrete version(i.e `0.0.1`). We aim to keep userland-apis relatively stable, but
```
Comment (hrhino) [inline README.md]: wee nit, missing space and period, i.e. "version (i.e. `0.0.1`)".

File: README.md [hunk 3]
```diff
@@ -20,6 +20,21 @@ For the current progress, please refer to the [RoadMap](https://github.com/jmcar
...
+### Note on milestones:
+Our Notation for versions is:
+```
+X.X.X
+^ ^ ^____Minor
+| |______Major
+|________Complete redesign (i.e scalaz 7 vs 8)  
+```
+
+All `x.x.x-Mx` releases are milestone releases. Thus, we do not guarantee binary compatibility or no api-breaking until
```
Comment (hrhino) [inline README.md]: "no api-breakage" while I'm being a pedant

File: README.md [hunk 4]
```diff
@@ -20,6 +20,21 @@ For the current progress, please refer to the [RoadMap](https://github.com/jmcar
...
+a concrete version(i.e `0.0.1`). We aim to keep userland-apis relatively stable, but 
+internals shift as we find better/more performant abstractions.
+
+We will guarantee compat between minor versions (i.e 0.0.1 => 0.0.2) but not major versions (0.0.1 => 0.1.0)
```
Comment (hrhino) [inline README.md]: Maybe just say "compatibility"?

**Pull request merged**